src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -271,11 +297,18 @@ ss::future<try_abort_reply> tx_gateway_frontend::try_abort_locally(
     }
 
     auto reply = co_await container().invoke_on(
```

Comment:
is it the case that you'd want to also enter the gate on the current core in addition to the core on which do_try_abort is being run? looks like the later is happening, but no the former.

Reply:
not sure about it, all the work happens on the target node so don't see a reason to take it

Reply:
i was assuming that you are trying to protect against races with shutdown? you have fiber running on the current core (the entire function / context executing `auto reply = co_await container().invoke_on(`), as well as the stuff on the other core.

Reply:
without knowing the precise race that is occurring its hard to say. it seems like the current approach is "add more gates" (which is pretty reasonable, really), but in that case it's worth asking why there aren't more gates :P

Reply:
In all instances of the problem it appears when a `tm_stm` is closed while there is an ongoing transaction. `tx_gateway_frontend` check if a `tm_stm` exists only in the beginning of a transaction so if `tm_stm` is deleted we may get the memory access violation. `tx_gateway_frontend` uses `tm_stm` only on the target node so it's ok to put gate after `container.invoke_on`

Reply:
i'm not saying it's incorrect to take a gate on the target core, i'm asking if there needs to _also_ be a gate taken on the source core.

Reply:
in the logs it looks like the tx_gateway_frontend is also disappearing, or might be, based on the first log line below

```
DEBUG 2022-09-20 17:38:17,466 [shard 1] tx - tx_gateway_frontend.cc:147 - Asked to stop tx coordinator
TRACE 2022-09-20 17:38:17,467 [shard 1] tx - rm_partition_frontend.cc:516 - received name:commit_tx, ntp:{kafka/tx-topic/0}, pid:{producer_identity: id=1, epoch=1}, tx_seq:2, ec:cluster::tx_errc:0
DEBUG 2022-09-20 17:38:17,467 [shard 1] rpc - transport.cc:250 - could not parse header from server: {host: docker-rp-13, port: 33145}
TRACE 2022-09-20 17:38:17,467 [shard 1] tx - tx_gateway_frontend.cc:50 - released_lock name:end_txn, tx_id:{test-tx-id}
```

in which case, i would seem like we need to also protect access to the frontend instance.

if shutdown occurred in this loop

```
    while (!aborted && !shard && 0 < retries--) {
        aborted = !co_await sleep_abortable(delay_ms, _as);
        shard = _shard_table.local().shard_for(model::tx_manager_ntp);
    }
```

then when it popped out and continued executing below

```
    auto reply = co_await container().invoke_on(
```

container() might be a UAF?

Reply:
it was happening because a spawned fiber wasn't holding a tm_stm's gate open, fixed with the last commit